test/include/test.hpp

```diff
@@ -402,6 +403,29 @@ auto within_abs(T px, U py, double ptol = 1e-6f)
         px, py, ptol);
 }
 
+template <class Iterator1, class Iterator2>
+bool glob_match(Iterator1 start, Iterator1 last, Iterator2 pattern_start, Iterator2 pattern_last)
+{
+    std::tie(start, pattern_start) =
+        std::mismatch(start, last, pattern_start, pattern_last, [](auto c, auto m) {
+            if(m == '?')
```

Comment:
Why would a match to '?' return a true here? It should behave like '*', but just for one character.

Reply:
We need a loop for `*` which `std::mismatch` can't handle which is why we bail here.

Reply:
Suggestion: please document what is this function trying to do here.

Reply:
Added a comment.